source/dub/packagesupplier.d

```diff
@@ -179,6 +186,11 @@ class RegistryPackageSupplier : PackageSupplier {
 		m_metadataCacheDirty = false;
 	}
 
+    Json searchForPackages(string[] names){
+        import std.array : join;
+        auto url = m_registryUrl ~ Path("api/search?q=" ~ names.join(","));
```

Comment:
When I went searching for what was encoding, I couldn't find any of the current uses of download() encoding the strings. To be safe however, I went and encoded it on the dub end using std.uri.encodeComponent (decoding on the dub-registry end with std.uri.decodeComponent).
Have tested and this works.